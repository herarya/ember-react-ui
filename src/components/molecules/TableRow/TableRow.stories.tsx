import type { Meta, StoryObj } from '@storybook/react';

import TableRow from './TableRow';
import TableDataCell from '../../atoms/TableDataCell/TableDataCell';

const meta: Meta<typeof TableRow> = {
  title: 'Ember UI/Molecules/TableRow',
  component: TableRow,
  tags: ['autodocs']
};


export default meta;

type Story = StoryObj<typeof TableRow>;

// fix Warning: validateDOMNesting(...): <td> cannot appear as a child of <div>.
const TableRoWTemplate: Story = {
  render: ({ ...args }) => {
    return (
      <table>
        <tbody>
          <TableRow {...args}>
              <TableDataCell variant="default" align='left'>Table Cell Content</TableDataCell>
          </TableRow>
        </tbody>
      </table>
    );
  },
};


export const Default: Story = {
    ...TableRoWTemplate,
    args: { },
};

export const Selected: Story = {
  ...TableRoWTemplate,
  args: {
    selected : true,
  },
};