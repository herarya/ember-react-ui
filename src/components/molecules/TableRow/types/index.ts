import { ReactNode } from 'react';

export type TableRowProps = {
  /**
   * The indicated as selected  of row;
   */
  selected?: boolean;

  /**
   * The content of cell
   */
  children: ReactNode;
};
