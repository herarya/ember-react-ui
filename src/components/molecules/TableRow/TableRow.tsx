import TableRowStyled from './styles/tablerow';
import { TableRowProps } from './types';

const TableRow: React.FC<TableRowProps> = ({ selected = false, children }: TableRowProps) => {
  return <TableRowStyled selected={selected}>{children}</TableRowStyled>;
};

export default TableRow;
