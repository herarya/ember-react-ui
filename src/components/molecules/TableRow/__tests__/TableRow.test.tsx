import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import * as stories from '../TableRow.stories';

const { Default , Selected } = composeStories(stories);

test('should render Row', () => {
    const container = render(<Default />);
    expect(container).toMatchSnapshot();
});


test('should render Row Selected', () => {
    const container = render(<Selected />);
    expect(container).toMatchSnapshot();
});

