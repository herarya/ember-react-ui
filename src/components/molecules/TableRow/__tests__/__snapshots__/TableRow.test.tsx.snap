// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Row 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table>
        <tbody>
          <tr
            class="sc-aXZVg bNeWin"
          >
            <td
              class="sc-gEvEer jzRORs"
            >
              Table Cell Content
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table>
      <tbody>
        <tr
          class="sc-aXZVg bNeWin"
        >
          <td
            class="sc-gEvEer jzRORs"
          >
            Table Cell Content
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render Row Selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table>
        <tbody>
          <tr
            class="sc-aXZVg cSaLre"
          >
            <td
              class="sc-gEvEer jzRORs"
            >
              Table Cell Content
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table>
      <tbody>
        <tr
          class="sc-aXZVg cSaLre"
        >
          <td
            class="sc-gEvEer jzRORs"
          >
            Table Cell Content
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
